### PR TITLE
Use MUI Dialog in favor of Fluent UI Dialog

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
Minimal visual changes (see storybook diffs for full differences)

**Description**
Replace fluentui Dialog components with @mui equivalents

### CODE REVIEW

File: packages/studio-base/package.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
     "@types/uuid": "8.3.4",
     "@types/wicg-file-system-access": "2020.9.5",
     "@types/ws": "8.5.2",
-    "@wojtekmaj/enzyme-adapter-react-17": "0.6.6",
+    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
```
Comment (jtbandes) [inline packages/studio-base/package.json:109]: what's this about?

File: packages/studio-base/src/components/HoverableIconButton.tsx [hunk 2]
```diff
@@ -8,23 +8,37 @@ import { forwardRef, useCallback, useEffect, useState } from "react";
...
   icon: React.ReactNode;
   activeIcon?: React.ReactNode;
-} & Omit<IconButtonProps, "children">;
+  color?: IconButtonProps["color"];
+  activeColor?: IconButtonProps["color"];
+} & Omit<IconButtonProps, "children" | "color">;
 
 const HoverableIconButton = forwardRef<HTMLButtonElement, Props>((props, ref) => {
-  const { icon, activeIcon, ...rest } = props;
+  const { icon, activeIcon, color, activeColor, onMouseLeave, onMouseEnter, ...rest } = props;
```
Comment (jtbandes) [inline packages/studio-base/src/components/HoverableIconButton.tsx:16]: What's the point of accepting enter/leave handlers as props? Looks like that is not being used anywhere?

File: packages/studio-base/src/components/LayoutBrowser/UnsavedChangesPrompt.tsx [hunk 3]
```diff
@@ -105,62 +86,63 @@ export function UnsavedChangesPrompt({
...
+        <DialogContent>
+          <Stack gap={2} style={{ minHeight: 180 }}>
+            <RadioGroup defaultValue="discard" onChange={handleChoiceGroupChange}>
+              <FormControlLabel value="discard" label="Discard changes" control={<Radio />} />
+              <FormControlLabel
+                value="overwrite"
+                label={[
+                  `Update team layout “${layout.name}”`,
+                  !isOnline && "(unavailable while offline)",
+                ].join(" ")}
```
Comment (jtbandes) [inline packages/studio-base/src/components/LayoutBrowser/UnsavedChangesPrompt.tsx:101]: Maybe add filter(Boolean)? As is, this change adds a trailing space in the case where isOnline=true.

**Review by jtbandes — APPROVED** [review]
- Feels pretty low contrast with no shadow, especially in dark mode:
  image
  image

- Looks like the Open dialog has not been converted yet?

- Some text alignment issue in the MarkerArray settings: 
  image

- Are we planning to use some kind of mui tooltip component in the future? The native `title` feels out of place because it's slow to appear and doesn't match our color scheme:
  image

File: packages/studio-base/package.json (hunk 1 above)
Comment (2metres) [reply packages/studio-base/package.json:109]: There is a very specific enzyme error that happens when trying to register a click event on a MUI (emotion) button that seems to be resolved by this version bump

File: packages/studio-base/src/components/HoverableIconButton.tsx [hunk 4]
```diff
@@ -8,23 +8,37 @@ import { forwardRef, useCallback, useEffect, useState } from "react";
...
+      if (onMouseEnter != undefined) {
+        onMouseEnter(event);
+      }
+      if (props.disabled === true) {
+        return;
+      }
+      setHovered(true);
+    },
+    [onMouseEnter, props.disabled],
+  );
```
Comment (2metres) [inline packages/studio-base/src/components/HoverableIconButton.tsx:31]: @jtbandes it's used here

**Comment by 2metres** [comment]
> Looks like the Open dialog has not been converted yet?

Yes, I've got a [separate PR](https://github.com/foxglove/studio/pull/3774) specificallty targeting the OpenDialog

**Comment by 2metres** [comment]
@jtbandes 

> Are we planning to use some kind of mui tooltip component in the future? The native title feels out of place because it's slow to appear and doesn't match our color scheme

Eventually yes, but for now i'm removing the use of useTooltip as it relies on Fluent UI. I'd be open to PRs that implement something similar using the MUI tooltip component.

File: packages/studio-base/src/components/HoverableIconButton.tsx (hunk 4 above)
Comment (jtbandes) [reply packages/studio-base/src/components/HoverableIconButton.tsx:31]: Yeah but are any users of HoverableIconButton actually passing these in?

**Pull request merged**